### PR TITLE
new plugin API update

### PR DESCRIPTION
**Pull request by milyin**
Update according to https://github.com/eclipse-zenoh/zenoh/pull/583

### CODE REVIEW

File: README.md
Comment (JEnoch) [inline README.md:1]: I suggest to further simplify this paragraph it instructing to just run `cargo build --release` (without `-p ...`) and describing the 2 resulting artifacts (lib plugin and bridge exe).

File: zenoh-plugin-dds/src/lib.rs
Comment (JEnoch) [inline zenoh-plugin-dds/src/lib.rs:1]: Should the call to `declare_plugin!()` have a `#[cfg(feature = "no_mangle")]` as in https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds/pull/51 ?

**Review by JEnoch — CHANGES_REQUESTED** [review]

File: zenoh-plugin-dds/src/lib.rs
Comment (milyin) [reply zenoh-plugin-dds/src/lib.rs:1]: Fixed. Added issue about this for further investigation https://github.com/eclipse-zenoh/zenoh/issues/751

File: README.md
Comment (milyin) [reply README.md:1]: done

**Pull request merged**